lua/customroles/missionary.lua

```diff
@@ -0,0 +1,149 @@
+local hook = hook
+local player = player
+
+local AddHook = hook.Add
+local PlayerIterator = player.Iterator
+
+local ROLE = {}
+
+ROLE.nameraw = "missionary"
+ROLE.name = "Missionary"
+ROLE.nameplural = "Missionaries"
+ROLE.nameext = "a Missionary"
+ROLE.nameshort = "mis"
+ROLE.team = ROLE_TEAM_DETECTIVE
+
+-- TODO: Missionary role descriptions
+ROLE.desc = [[You are {role}!]]
+ROLE.shortdesc = ""
+
+ROLE.loadout = {"weapon_mis_proselytizer"}
+
+local missionary_prevent_monk = CreateConVar("ttt_missionary_prevent_monk", "1", FCVAR_REPLICATED, "Whether to only spawn the Missionary when there isn't already a Monk, Zealot, or Hermit in the round", 0, 1)
+ROLE.selectionpredicate = function()
+    if not ROLE_SOULBOUND then return false end
+    if not missionary_prevent_monk:GetBool() then return true end
+
+    for _, p in PlayerIterator() do
+        if p:IsMonk() or p:IsZealot() or p:IsHermit() then
+            return false
+        end
+    end
+    return true
+end
+
+ROLE.convars = {
+    {
+        cvar = "ttt_missionary_proselytizer_time",
+        type = ROLE_CONVAR_TYPE_NUM,
+        decimal = 0
+    },
+    {
+        cvar = "ttt_missionary_announce_proselytize",
+        type = ROLE_CONVAR_TYPE_DROPDOWN,
+        choices = {"Don't announce", "Announce as Missionary", "Announce as Marshal"},
+        isNumeric = true
+    },
+    {
+        cvar = "ttt_missionary_prevent_monk",
+        type = ROLE_CONVAR_TYPE_BOOL
+    }
+}
+
+ROLE.translations = {
+    ["english"] = {
+        ["mis_proselytizer_help_pri"] = "Hold {primaryfire} to deputize a player.",
+        ["missionary_proselytize_announce"] = "{amissionary} has proselytized {target}",
+        ["ev_missionary_proselytize"] = "{target} was proselytized by {missionary}"
+    }
+}
+
+RegisterRole(ROLE)
+
+local missionary_announce_proselytize = CreateConVar("ttt_missionary_announce_proselytize", "1", FCVAR_REPLICATED, "How a player being proselytized will be announced to everyone", 0, 2)
+
+MISSIONARY_ANNOUNCE_NONE = 0
+MISSIONARY_ANNOUNCE_AS_MISSIONARY = 1
+MISSIONARY_ANNOUNCE_AS_MARSHAL = 2
+
+if SERVER then
+    AddCSLuaFile()
+
+    ------------
+    -- EVENTS --
+    ------------
+
+    AddHook("Initialize", "Missionary_Initialize", function()
+        EVENT_PROSELYTIZED = GenerateNewEventID(ROLE_MISSIONARY)
+    end)
+end
+
+if CLIENT then
+    ------------------
+    -- TRANSLATIONS --
+    ------------------
+
+    AddHook("Initialize", "Missionary_Translations_Initialize", function()
+        -- This needs to happen after the Monk, Zealot and Hermit have been registered so we can't put it in ROLE.translations
+        LANG.AddToLanguage("english", "mis_proselytizer_help_sec", "The target player will become " .. ROLE_STRINGS_EXT[ROLE_MONK] .. ", " .. ROLE_STRINGS[ROLE_ZEALOT] .. ", or " .. ROLE_STRINGS[ROLE_HERMIT])
```

Comment:
You can pass parameters to `SWEP:AddHUDHelp` as the 4th parameter so you can register this in `ROLE.translations` and then pass the role strings there.